homeassistant/components/alarm_control_panel/alarmdecoder.py

```diff
@@ -42,22 +71,63 @@ def async_added_to_hass(self):
             SIGNAL_PANEL_MESSAGE, self._message_callback)
 
     def _message_callback(self, message):
+        do_update = False
+
         if message.alarm_sounding or message.fire_alarm:
             if self._state != STATE_ALARM_TRIGGERED:
```

Comment:
You can remove all these checks of the old state. They don't add any value. Also remove the check if an update should be scheduled. Just schedule the update. There will only be a state update if the state has changed anyway.